actionpack/lib/action_dispatch/middleware/static.rb

```diff
@@ -94,9 +98,9 @@ def gzip_file_path(path)
   # produce a directory traversal using this middleware. Only 'GET' and 'HEAD'
   # requests will result in a file being returned.
   class Static
-    def initialize(app, path, cache_control=nil)
+    def initialize(app, path, cache_control = nil, headers = {})
       @app = app
-      @file_handler = FileHandler.new(path, cache_control)
+      @file_handler = FileHandler.new(path, cache_control, headers)
```

Comment:
(That'd mean deprecating `config.static_cache_control` as well, transitioning to a headers config setting.)